javaslang/src/main/java/javaslang/collection/Array.java

```diff
@@ -593,7 +592,9 @@ public T get(int index) {
                 list.add(t);
             }
         }
-        if (list.size() == back.length) {
+        if (list.isEmpty()) {
```

Comment:
changed most `filter` methods to return empty, if possible.
This affects many other methods also.